Add mysensors IR switch device and service

MartinHjelmare opened the pull request:
**Description:**
This PR makes it possible to change an IR code for a MySensors IR sender device and turn that on to send the code. After turning on to send the code the device should be turned off.
- Add MySensorsIRSwitch as child class to MySensorsSwitch.
- Add platform specific service mysensors_send_ir_code. Only call
  device method in service function if device is IR device.
- Add service and required attribute to state helper to support scenes.
- Move V_IR_SEND type from sensor.mysensors to switch.mysensors
  platform.
- Populate switch.services.yaml with service descriptions.

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):**
https://github.com/home-assistant/home-assistant.io/pull/556

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
- [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If code communicates with devices:
- [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**

MartinHjelmare committed:
Add mysensors IR switch device and service

* Add MySensorsIRSwitch as child class to MySensorsSwitch.
* Add platform specific service mysensors_send_ir_code. Only call
	device method in service function if device is IR device.
* Add service and required attribute to state helper to support scenes.
* Move V_IR_SEND type from sensor.mysensors to switch.mysensors
	platform.
* Populate switch.services.yaml with service descriptions.

balloob commented:
So this looks fine, I don't see anything from looking at the code that this would not work 🐬 

You can merge it if you think it's ready.

MartinHjelmare committed:
Fix check of entity_id in service function

Since multiple entity_ids can be passed as service data, and the
entity_id service attribute is forced to a list by the service
validation schema, the check in the service function should iterate
over any entity ids.

MartinHjelmare commented:
Fixed the check of entity_id in service function. Docs added. Merging.
